ApiTests/Tests/UI/DynamicControlsTest.cs

```diff
@@ -1,24 +1,36 @@
 ﻿using ApiTests.Tests.UI;
 using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using SeleniumExtras.WaitHelpers;
 using TestProject.Tests.Pages;
 using TestProject.Utils;
 
 namespace TestProject.Tests.UI
 {
     internal class DynamicControlsTests : BaseTest
     {
+        readonly MainPage mainPage = new MainPage();
         private static readonly By enableBtn = By.XPath(string.Format(XpathPatterns.preciseTextXpath, "Enable"));
+        private static readonly By disableBtn = By.XPath(string.Format(XpathPatterns.preciseTextXpath, "Disable"));
+        private static readonly By inputField = By.XPath(string.Format(XpathPatterns.typeText));
+        private static readonly string randomValue = Guid.NewGuid().ToString();
 
         [Test]
         public void DynamicControlsTest()
         {
-            MainPage mainPage = new MainPage();
             mainPage.ClickOnDynamicControl();
+            WebDriverWait wait = new WebDriverWait(Browser.GetDriver(), TimeSpan.FromSeconds(100));
+            wait.Until(ExpectedConditions.ElementToBeClickable(enableBtn));
             Browser.GetDriver().FindElement(enableBtn).Click();
+            wait.Until(ExpectedConditions.ElementIsVisible(disableBtn));
             //assert input is enabled
+            Assert.That(Browser.GetDriver().FindElement(disableBtn).Enabled, Is.True);
```

Comment:
The good practice is to put a message in an assertion. I can see it in other places though, so this one was probably missed